packages/picasso.js/src/core/geometry/geopolygon.js

```diff
@@ -0,0 +1,225 @@
+   * @param {Array} [input.vertices=[]] Vertices are represented as an array of arrays of points.
+   */
+  set({ vertices = [] } = {}) {
+    this.type = 'geopolygon';
+    this.vertices = vertices.slice();
+    this.numPolygons = this.vertices.length;
+    this.polygons = [];
+    this.xMin = NaN;
+    this.yMin = NaN;
+    this.xMax = NaN;
+    this.yMax = NaN;
+
+    for (let i = 0; i < this.numPolygons; i++) {
+      removeDuplicates(this.vertices[i]);
+      if (this.vertices[i].length > 2) {
+        close(this.vertices[i]);
+      }
+      this.polygons[i] = createPolygon({ vertices: this.vertices[i] });
+      this.xMin = isNaN(this.xMin) ? this.polygons[i].xMin : Math.min(this.xMin, this.polygons[i].xMin);
+      this.xMax = isNaN(this.xMax) ? this.polygons[i].xMax : Math.max(this.xMax, this.polygons[i].xMax);
+      this.yMin = isNaN(this.yMin) ? this.polygons[i].yMin : Math.min(this.yMin, this.polygons[i].yMin);
+      this.yMax = isNaN(this.yMax) ? this.polygons[i].yMax : Math.max(this.yMax, this.polygons[i].yMax);
+    }
+
+    this._bounds = null;
+    this._boundingRect = null;
+  }
+
+  /**
+   * Check if a point is inside the area of the geopolygon.
+   * Supports convex, concave and self-intersecting polygons (filled area).
+   * @param {point} point
+   * @returns {boolean} True if there is an intersection, false otherwise
+   */
+  containsPoint(point) {
+    return testGeoPolygonPoint(this, point);
+  }
+
+  /**
+   * Check if circle is inside the area of the polygon.
+   * Supports convex, concave and self-intersecting polygons (filled area).
+   * @param {circle} circle
+   * @returns {boolean} True if there is an intersection, false otherwise
+   */
+  intersectsCircle(circle) {
+    return testCircleGeoPolygon(circle, this);
+  }
+
+  /**
+   * @param {point[]} points - Line start and end point as an array of points
+   * @returns {boolean} True if there is an intersection, false otherwise
+   */
+  intersectsLine(points) {
+    return testGeoPolygonLine(this, pointsToLine((points)));
+  }
+
+  /**
+   * @param {point[]} points - Rect vertices as an array of points
+   * @returns {boolean} True if there is an intersection, false otherwise
+   */
+  intersectsRect(points) {
+    return testGeoPolygonRect(this, pointsToRect(points));
+  }
+
+  /**
+   * Check if polygon intersects another polygon.
+   * Supports convex, concave and self-intersecting polygons (filled area).
+   * @param {Polygon} polygon
+   * @returns {boolean} True if there is an intersection, false otherwise
+   */
+  intersectsPolygon(polygon) {
+    const rect1 = this.boundingRect();
+    const rect2 = polygon.boundingRect();
+    if (!testRectRect(rect1, rect2)) {
+      return false;
+    }
+    if (testRectContainsRect(rect2, rect1)) {
+      return testPolygonGeoPolygon(polygon, this);
+    }
+    return testGeoPolygonPolygon(this, polygon);
+  }
+
+  /**
+   * Get the points
+   * @returns {point[]}
+   */
+  points() {
```

Comment:
`points` API should return an array of points. The current use case for it is when debugging collision logic. I suggest you just return an empty array to avoid exceptions.

Reply:
This is the same as polygon.

Reply:
It's still invalid.